app/assets/javascripts/discourse/components/badge-title.js.es6

```diff
@@ -0,0 +1,25 @@
+import { ajax } from 'discourse/lib/ajax';
+import BadgeSelectController from 'discourse/mixins/badge-select-controller';
+
+export default Ember.Component.extend(BadgeSelectController, {
+  classNames: ["badge-title"],
+
+  actions: {
+    save() {
+      this.setProperties({ saved: false, saving: true });
+
+      ajax(this.get('user.path') + "/preferences/badge_title", {
+        type: "PUT",
+        data: { user_badge_id: this.get('selectedUserBadgeId') }
+      }).then(() => {
+        this.setProperties({
+          saved: true,
```

Comment:
You should declare the attributes you will be using in the component with default values. Looks like `saved: false, saving:false` - Javascript is much faster when it knows the shape of objects in advance